scripts/adlsgen2setup.py

```diff
@@ -0,0 +1,166 @@
+import argparse
+import asyncio
+import os
+
+import aiohttp
+from azure.core.credentials_async import AsyncTokenCredential
+from azure.identity.aio import AzureDeveloperCliCredential
+from azure.storage.filedatalake.aio import (
+    DataLakeDirectoryClient,
+    DataLakeServiceClient,
+)
+
+
+class AdlsGen2Setup:
+    async def __aenter__(self):
+        self.service_client = DataLakeServiceClient(
+            account_url=f"https://{self.storage_account_name}.dfs.core.windows.net", credential=self.credentials
+        )
+        await self.service_client.__aenter__()
+        return self
+
+    async def __aexit__(self, *args, **kwargs):
+        await self.service_client.__aexit__(*args, **kwargs)
+
+    def __init__(
+        self,
+        data_directory: str,
+        storage_account_name: str,
+        filesystem_name: str,
+        security_enabled_groups: bool,
+        credentials: AsyncTokenCredential,
+        verbose: bool = False,
+    ):
+        self.data_directory = data_directory
+        self.storage_account_name = storage_account_name
+        self.filesystem_name = filesystem_name
+        self.credentials = credentials
+        self.security_enabled_groups = security_enabled_groups
+        self.verbose = verbose
+
+    async def run(self):
+        token_result = await self.credentials.get_token("https://graph.microsoft.com/.default")
+        self.graph_headers = {"Authorization": f"Bearer {token_result.token}"}
+        if self.verbose:
+            print("Ensuring gptkbcontainer exists...")
```

Comment:
I think it'd be nicer to move away from print() statements and use logging module instead, these could be logging.info() or logging.debug(). Then you can set overall logging level to decide when to see them.  (And if you ran this on a server, it'd be easier to pipe logs somewhere)